src/components/SidePanels/TabPanels/SideTabsPanels.tsx

```diff
@@ -0,0 +1,433 @@
+        top,
+        left,
+        relativeTop: top / viewportSize.current.height * 100,
+        relativeLeft: left / viewportSize.current.width * 100,
+        storedLeft: undefined,
+        storedTop: undefined,
+        detached,
+        maxHeight,
+        alignment: detached ? undefined : panel.alignment,
+      });
+    });
+  }, [updatePanel, checkSnap, panelData]);
+
+  const onResizeStart = useCallback(() => {
+    setResizing(() => true);
+  }, []);
+
+  const onResizeEnd = useCallback(() => {
+    setResizing(() => false);
+  }, []);
+
+  const findPanelsOnSameSide = useCallback((panelAlignment : string) => {
+    return Object.keys(panelData)
+      .filter((panelName) => panelData[panelName as string]?.alignment === panelAlignment);
+  }, [panelData]);
+
+
+  const onResize = useCallback((key: string, w: number, h: number, t: number, l: number) => {
+    const { left, top } = normalizeOffsets(key, t, l);
+    const maxHeight = viewportSize.current.height - top;
+
+    requestAnimationFrame(() => {
+      const panelsOnSameAlignment = findPanelsOnSameSide(panelData[key]?.alignment);
+
+      panelsOnSameAlignment.forEach(() => {
+        updatePanel(key, {
+          top,
+          left,
+          relativeTop: (top / viewportSize.current.height) * 100,
+          relativeLeft: (left / viewportSize.current.width) * 100,
+          storedLeft: undefined,
+          storedTop: undefined,
+          maxHeight,
+          width: clamp(w, DEFAULT_PANEL_WIDTH, panelMaxWidth),
+          height: clamp(h, DEFAULT_PANEL_HEIGHT, maxHeight),
+        });
+      });
+    });
+  }, [updatePanel, panelMaxWidth, panelData]);
+
+  const onSnap = useCallback((key: string) => {
+    setPositioning(false);
+
+    if (!localSnap.current) return;
+    const bboxData: Partial<PanelBBox> = {
+      alignment: localSnap.current as DroppableSide,
+      detached: false,
+    };
+
+    const firstPanelOnNewSideName = findPanelsOnSameSide(localSnap.current).filter(panelName => panelName !== key)?.[0];
+
+    if(firstPanelOnNewSideName) {
+      bboxData.width = clamp(panelData[firstPanelOnNewSideName as string]?.width, DEFAULT_PANEL_WIDTH, panelMaxWidth);
+    }
+    
+    updatePanel(key, bboxData);
+    setSnap(undefined);
+  }, [updatePanel]);
+
+  const eventHandlers: EventHandlers = useMemo(() => {
+    return {
+      onResize,
+      onResizeStart,
+      onResizeEnd,
+      onPositionChange,
+      onVisibilityChange,
+      onPositionChangeBegin,
+      onSnap,
+      transferTab,
+      createNewPanel,
+      setActiveTab,
+    };
+  }, [onResize, onResizeStart, onResizeEnd, onPositionChange, onVisibilityChange, onSnap, transferTab, createNewPanel, setActiveTab]);
+
+  const commonProps: CommonProps = useMemo(() => {
+    return {
+      ...eventHandlers,
+      root: rootRef,
+      regions,
+      selection: regions.selection,
+      currentEntity,
+    };
+  }, [eventHandlers, rootRef, regions, regions.selection, currentEntity]);
```

Comment:
same thing here, rootRef is a stable reference, it won't trigger dependency arrays.